x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/spatial/BinarySpatialFunction.java

```diff
@@ -80,6 +83,42 @@ public void writeTo(StreamOutput out) throws IOException {
         // The CRS type is re-resolved from the combination of left and right fields, and also not necessary to serialize
     }
 
+    /**
+     * Mark the function as expecting the specified fields to arrive as doc-values.
+     */
+    public abstract BinarySpatialFunction withDocValues(Set<FieldAttribute> attributes);
+
+    /**
+     * When performing local physical plan optimization, it is necessary to know if this function has a field attribute.
+     * This is because the planner might push down a spatial aggregation to lucene, which results in the field being provided
+     * as doc-values instead of source values, and this function needs to know if it should use doc-values or not.
+     */
+    public boolean hasFieldAttribute(Set<FieldAttribute> foundAttributes) {
```

Comment:
These methods pre-exist this PR, and were just moved from one class to another to fix an issue with the scope of the spatial filter pushdown. However, I see your point that they could be moved to the `SpatialDocValuesExtraction` class since that is the only place they are ever used. This could be done in the PR; or in another cleanup PR.

Reply:
> These methods pre-exist this PR
Yes, I've tracked down the history of those methods, but now I took my time to look over the code concerning spatial functions in detail.

Let's try doing the move in this PR 👍

Reply:
Done!